src/Http/Controllers/Requests/DefaultFormRequest.php

```diff
@@ -3,26 +3,86 @@
 namespace Weap\Junction\Http\Controllers\Requests;
 
 use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Support\Facades\Auth;
+use Spatie\MediaLibrary\MediaCollections\Models\Media;
+use Weap\Junction\Http\Utilities\MediaFile;
+use Weap\Junction\Models\MediaTemporaryUpload;
 
 class DefaultFormRequest extends FormRequest
 {
     /**
-     * Get the validation rules that apply to the request.
-     *
-     * @return array
+     * @return void
      */
-    public function rules(): array
+    protected function prepareForValidation()
     {
-        return method_exists($this->route()->controller, 'rules') ? $this->route()->controller->rules() : [];
+        if (class_exists(Media::class) && config('media-library.media_model')) {
+            $this->merge($this->prepareMedia($this->request->all()));
+        }
     }
 
     /**
-     * Get custom messages for validator errors.
-     *
-     * @return array
+     * @param array<string, mixed> $data
+     * @return array<string, mixed>
      */
-    public function messages(): array
+    private function prepareMedia(array $data): array
     {
-        return method_exists($this->route()->controller, 'messages') ? $this->route()->controller->messages() : [];
+        foreach ($data as $key => $value) {
+            if (! is_array($value)) {
+                continue;
+            }
+
+            if ($key !== '_media' || ! $this->isValidMediaArray($value)) {
+                $data[$key] = $this->prepareMedia($value);
+
+                continue;
+            }
+
+            $mediaArray = [];
+
+            foreach ($value as $collectionName => $mediaItems) {
+                foreach ($mediaItems as $mediaId) {
+                    $media = config('media-library.media_model')::find($mediaId);
+
+                    if (! $media || $media->model_type !== MediaTemporaryUpload::class) {
+                        $mediaArray[$collectionName][] = $mediaId;
+
+                        continue;
+                    }
+
+                    abort_if(Auth::user()->id !== $media->model->created_by_user_id, 404);
```

Comment:
- Might be bit cleaner to use `Auth::id()`
- Should this be `403` because it seems more like an authorization issue?